android/src/main/java/com/reactnativestripesdk/pushprovisioning/PushProvisioningProxy.kt

```diff
@@ -0,0 +1,109 @@
+package com.reactnativestripesdk
+
+import android.app.Activity
+import android.app.Activity.RESULT_OK
+import android.content.Intent
+import android.util.Log
+import com.facebook.react.bridge.*
+import com.reactnativestripesdk.pushprovisioning.AddToWalletButtonView
+import com.reactnativestripesdk.pushprovisioning.EphemeralKeyProvider
+import com.reactnativestripesdk.pushprovisioning.TapAndPayProxy
+import com.stripe.android.pushProvisioning.PushProvisioningActivity
+import com.stripe.android.pushProvisioning.PushProvisioningActivityStarter
+
+
+object PushProvisioningProxy {
+  private const val TAG = "StripePushProvisioning"
+  private var description = "Added by Stripe"
+  private var tokenRequiringTokenization: ReadableMap? = null
+
+  fun getApiVersion(): String {
+    return try {
+      Class.forName("com.stripe.android.pushProvisioning.PushProvisioningActivity")
+      PushProvisioningActivity.API_VERSION
+    } catch (e: Exception) {
+      Log.e(TAG, "PushProvisioning dependency not found")
+      ""
+    }
+  }
+
+  fun invoke(
+          context: ReactApplicationContext,
+          view: AddToWalletButtonView,
+          cardDescription: String,
+          ephemeralKey: String,
+          token: ReadableMap?
+  ) {
+    try {
+      Class.forName("com.stripe.android.pushProvisioning.PushProvisioningActivityStarter")
+      description = cardDescription
+      tokenRequiringTokenization = token
+      createActivityEventListener(context, view)
+      DefaultPushProvisioningProxy().beginPushProvisioning(
+        context.currentActivity!!,
+        description,
+        EphemeralKeyProvider(ephemeralKey)
+      )
+    } catch (e: Exception) {
+      Log.e(TAG, "PushProvisioning dependency not found")
```

Comment:
I wasn't able to see this log in the console when I removed the dependency and pressed add to gpay, am I missing something?

Reply:
since [you're not passing an `ephemeralKey`](https://github.com/stripe/stripe-react-native/pull/889#discussion_r856429833), I bet you're actually receiving [this error via the `onComplete` hook?](https://github.com/stripe/stripe-react-native/blob/charliecruzan-push-provisioning/android/src/main/java/com/reactnativestripesdk/pushprovisioning/AddToWalletButtonView.kt#L48)

Reply:
so the `.invoke` function isn't actually called.